src/caffeine/caffeine.c

```diff
@@ -194,7 +190,7 @@ void caf_co_max(CFI_cdesc_t* a_desc, int result_image, int* stat, char* errmsg,
     case CFI_type_float:            a_type = GEX_DT_FLT; break;
     case CFI_type_double:           a_type = GEX_DT_DBL; break;
     default:
-      set_stat_errmsg_or_abort(stat, errmsg, UNRECOGNIZED_TYPE, "");
+      set_stat_errmsg_or_abort(NULL, NULL, UNRECOGNIZED_TYPE, "");
```

Comment:
IIUC the call above is equivalent to `gasnett_fatalerror("")`, which is better than `abort()` but still not great: the message doesn't give a hint about what's wrong, due to the level of call indirection the failing line number won't even point to this line as the root cause, nor does it give a hint about what unrecognized type was passed.

```suggestion
      gasnett_fatalerror("Unrecognized type: %d", (int)a_desc->type);
```